includes/html/pages/device/ports/xdsl.inc.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+echo "<div style='margin: 5px;'><table border=0 cellspacing=0 cellpadding=5 width=100%>";
+
+echo '<tr><th>Port</th><th>Traffic</th><th>Sync Speed</th><th>Attainable Speed</th><th>Attenuation</th><th>SNR Margin</th><th>Output Powers</th></tr>';
+$i = '0';
+
+$ports = DeviceCache::getPrimary()->ports()->join('ports_adsl', 'ports.port_id', 'ports_adsl.port_id')
```

Comment:
These can probably use the relationships now:

DeviceCache::getPrimary()->ports()->with(['portsAdsl', 'portsVdsl'])->orderby('ports.ifIndex', 'ASC')

This will change the access inside the include though as instead of the properties being in the base model, they will be  under $port->portsAdsl and $port->portsVdsl respectively.

Reply:
I did try to play with that one without success for now. Will continue later.